Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -172,87 +175,101 @@ public void Serialize<T>(IIonWriter writer, T item)
                 return;
             }
 
-            if (item is int)
+            if (type.IsAssignableTo(typeof(int)))
             {
-                new IonIntSerializer().Serialize(writer, Convert.ToInt32(item));
+                var serializer = this.GetCustomPrimitiveSerializer<int>() ?? new IonIntSerializer();
+                serializer.Serialize(writer, Convert.ToInt32(item));
                 return;
             }
 
-            if (item is long)
+            if (type.IsAssignableTo(typeof(long)))
             {
-                new IonLongSerializer().Serialize(writer, Convert.ToInt64(item));
+                var serializer = this.GetCustomPrimitiveSerializer<long>() ?? new IonLongSerializer();
+                serializer.Serialize(writer, Convert.ToInt64(item));
                 return;
             }
 
-            if (item is float)
+            if (type.IsAssignableTo(typeof(float)))
             {
-                new IonFloatSerializer().Serialize(writer, Convert.ToSingle(item));
+                var serializer = this.GetCustomPrimitiveSerializer<float>() ?? new IonFloatSerializer();
+                serializer.Serialize(writer, Convert.ToSingle(item));
                 return;
             }
 
-            if (item is double)
+            if (type.IsAssignableTo(typeof(double)))
             {
-                new IonDoubleSerializer().Serialize(writer, Convert.ToDouble(item));
+                var serializer = this.GetCustomPrimitiveSerializer<double>() ?? new IonDoubleSerializer();
+                serializer.Serialize(writer, Convert.ToDouble(item));
                 return;
             }
 
-            if (item is decimal)
+            if (type.IsAssignableTo(typeof(decimal)))
             {
-                new IonDecimalSerializer().Serialize(writer, Convert.ToDecimal(item));
+                var serializer = this.GetCustomPrimitiveSerializer<decimal>() ?? new IonDecimalSerializer();
+                serializer.Serialize(writer, Convert.ToDecimal(item));
                 return;
             }
 
-            if (item is BigDecimal)
+            if (type.IsAssignableTo(typeof(BigDecimal)))
             {
-                new IonBigDecimalSerializer().Serialize(writer, (BigDecimal)(object)item);
+                var serializer = this.GetCustomPrimitiveSerializer<BigDecimal>() ?? new IonBigDecimalSerializer();
+                serializer.Serialize(writer, (BigDecimal)(object)item);
                 return;
             }
 
-            if (item is byte[])
+            if (type.IsAssignableTo(typeof(byte[])))
             {
-                new IonByteArraySerializer().Serialize(writer, (byte[])(object)item);
+                var serializer = this.GetCustomPrimitiveSerializer<byte[]>() ?? new IonByteArraySerializer();
+                serializer.Serialize(writer, (byte[])(object)item);
                 return;
             }
 
-            if (item is string)
+            if (type.IsAssignableTo(typeof(string)))

```

Comment:
`string` is sealed, therefore `is` is fine.

Reply:
Addressed.